processing/p5.js-sound
# Start audio context

following up on https://github.com/processing/p5.js-sound/pull/283

This pulls in Yotam Mann's [StartAudioContext](https://github.com/tambien/StartAudioContext).

I got a bit sidetracked looking at his new library [UnmuteButton](https://github.com/Tonejs/unmute) which would be another very cool solution. But it would involve upgrading to Tone 13.2, and shimming the unmute library (which uses `class`) to es5 or making p5.sound es6 friendly. If anyone wants to take this on that would be amazing.

In the meantime, this PR:

- adds a `userStartAudio` method
- adds a dependencey on StartAudioContext

To test:
- launch Chrome Canary (70+) from the terminal with the following `--disable-features` flags, i.e.
``` 
/Applications/Google\ Chrome\ Canary.app/Contents/MacOS/Google\ Chrome\ Canary
--disable-features=PreloadMediaEngagementData,AutoplayIgnoreWebAudio,MediaEngagementBypassAutoplayPolicies
```

Here's a little sketch to test—it should alert and remove the button after `userStartAudio` succeeds

```
function setup() {
  var ac = getAudioContext();

  ac.suspend().then(function() {
    var myButton = createButton('click to start audio');
    myButton.position(0, 0);

    userStartAudio(myButton, function() {
      alert(ac.state);
      myButton.remove();
    });
  });
}
```
